### PR TITLE
[MLIR] Visit Modules in the TranslationUnit scope

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Towards: https://github.com/lfortran/lfortran/issues/4319

### CODE REVIEW

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -999,7 +999,7 @@ RUN(NAME parameter_15 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran)
 
 RUN(NAME fortuno_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 
-RUN(NAME modules_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc wasm fortran)
+RUN(NAME modules_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc wasm fortran, mlir)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:1002]:
```suggestion
RUN(NAME modules_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc wasm fortran mlir)
```

**Review by certik — APPROVED** [review]
Looks good.

CI has to be fixed.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply integration_tests/CMakeLists.txt]: Yes.

**Comment by Thirumalai-Shaktivel** [comment]
Also, there was a failure, which needed a fix:
```
[...]
Error copying Fortran module "modules_01_a.mod".  Tried "MODULES_01_A.mod" and "modules_01_a.mod".
make[2]: *** [CMakeFiles/modules_01.dir/depend.make:7: CMakeFiles/modules_01.dir/modules_01_a.mod.stamp] Error 1
make[1]: *** [CMakeFiles/Makefile2:307: CMakeFiles/modules_01.dir/all] Error 2
make: *** [Makefile:101: all] Error 2
Command failed.
```

I fixed that here: dca8705d7aecd6b18e28ce5e0ed73ba6b695c1c5

**Comment by Thirumalai-Shaktivel** [comment]
As this PR is approved, I'm setting it to auto-merge.

**Comment by Thirumalai-Shaktivel** [comment]
Also, the failure is unrelated, so re-running it.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review!!

**Pull request merged**